### ISSUE TITLE
PDF export prefix and postfix

### ISSUE THREAD

**Issue by avitenberg** [angular-ui/ui-grid#4073]
I'm trying to display several lines of text above the exported table. Everything displayed in header is cut by the height of this header, and if I'm not mistaken I can set its size only for ALL pages, using code such as

exporterPdfCustomFormatter: function (docDefinition) {
                  docDefinition.pageMargins = [40, 200, 40, 40];
                  return docDefinition;
                },

But in fact I don't need such a huge margin for all pages, only for the first one where I display report title and parameters. And it's not correct even to regard it as "margin" or "header", it's not header, it's grid export prefix, it is not intended to be displayed in page header, it's a regular part of a document.

Using  exporterPdfTableStyle: {margin: [40, 200, 40, 40]}, it is possible to introduce an empty space before the grid, but it will remain empty, because there's no way to put anything in it.

Looks like the only way to handle is problem is by introducing two additional gridOptions, exporterPdfPrefix and exporterPdfPostfix. Is it a valid feature request or there's a better way to handle this problem? This situation is quite common (for example, when you need to display several report parameters at the beginning of the report), and support for this will add your pdf exporting a lot of flexibility.

**Comment by PaulL1** [comment]
Yes, this sounds valid, and a PR would be welcome.

**Comment by avitenberg** [comment]
Thank you very much for quick response. Will try to implement and do a pull request

**Comment by avitenberg** [comment]
I've found a way to do it without any changes to UI grid, with code like this

exporterPdfCustomFormatter: function (docDefinition) {
docDefinition.content= [{text: "My prefix"}].concat(docDefinition.content);
return docDefinition;
},

so it turned out that these properties exporterPdfPrefix and exporterPdfPostfix seems to be excessive, and I closed the pull request. But maybe it would be a good idea to include this treatment of prefixes to ui grid tutorial, and even remove other excessive exporter properties like exporterPdfPage size which can also be handled from within the exporterPdfCustomFormatter.

Thank you very much, and good luck with your excellent ui grid project!
